Mbed-TLS/mbedtls
# Refactor AES context to be shallow-copyable

## Description
Replaces use of a round key pointer in `mbedtls_aes_context` with a buffer offset, so that the structure is shallow-copyable. This retains support for alignment requirements with PadLock. A test is added to confirm a shallow-copy of the structure is valid. Fixes #2147.

## Status
**READY**

## Requires Backporting
Yes | NO 

## Todos


## Steps to test or reproduce
Initialise an `mbedtls_aes_context` structure, set encryption keys and then encrypt plaintext. Shallow-copy and then zero the original context, and then repeat encryption of the same plaintext using the copied context. The two ciphertexts should match, but will not prior to this change.
A similar test has been added in this PR.
